src/os/bluestore/BlueStore.h

```diff
@@ -512,10 +521,24 @@ class BlueStore : public ObjectStore,
     uint8_t  blob_depth;              /// blob overlapping count
     BlobRef blob;                     ///< the blob with our data
 
-    explicit Extent() {}
-    explicit Extent(uint32_t lo) : logical_offset(lo) {}
+    /// ctor for lookup only
+    explicit Extent(uint32_t lo) : logical_offset(lo) { }
+    /// ctor for delayed intitialization (see decode_some())
+    explicit Extent(Cache *cache) {
```

Comment:
IMHO specific blob assignment method that also increments cache counter(s) is more transparent and error prone, e.g.
void Extent::assign_blob(BlobRef b) {
  if(blob)
    blob->shared_blob->bc.cache->rm_extent();
  blob = b;
  if( blob )
    blob->shared_blob->bc.cache->add_extent();
}